Rev12 preparation (#4)
ietf-ccamp-wg/draft-ietf-ccamp-dwdm-if-param-yang: draft-ietf-ccamp-dwdm-if-param-yang.txt

Internet Engineering Task Force                       G. Galimberti, Ed.
Internet-Draft                                                         C
Intended status: Standards Track                     D. Hiremagalur, Ed.
Expires: 21 June 2025                                    G. Grammel, Ed.
                                                                 Juniper
                                                        R. Manzotti, Ed.
                                                                   Cisco
                                                          D. Breuer, Ed.
                                                     DEUTSCHE TELEKOM AG
                                                        18 December 2024

A YANG model to manage the optical interface parameters for an external

>                       transponder in a WDM network

Author: Updated the title as "A YANG data model to manage configurable DWDM optical interfaces"

                 draft-ietf-ccamp-dwdm-if-param-yang-12

Abstract

   This memo defines a Yang model related to the Optical Transceiver
   parameters characterising coherent 100G and above interfaces.  100G
   and above Transceivers support coherent modulation, multiple
   modulation formats, multiple FEC codes including some not yet
   specified (or in phase of specification by) [ITU-T_G.698.2] or any
   other ITU-T recommendation.  Use cases are described in [RFC7698].

   The Yang model defined in this memo can be used for Optical

>    Parameters monitoring and/or configuration of coherent interfaces.

Reviewer: Is this model only applicable to coherent interfaces?

>    Parameters monitoring and/or configuration of coherent interfaces.

Reviewer: Is this model only applicable to coherent interfaces?

     4.2.  Yang Module . . . . . . . . . . . . . . . . . . . . . . .   6
   5.  Security Considerations . . . . . . . . . . . . . . . . . . .  13
   6.  IANA Considerations . . . . . . . . . . . . . . . . . . . . .  14
   7.  Acknowledgements  . . . . . . . . . . . . . . . . . . . . . .  15
   8.  Contributors  . . . . . . . . . . . . . . . . . . . . . . . .  15
   9.  References  . . . . . . . . . . . . . . . . . . . . . . . . .  16
     9.1.  Normative References  . . . . . . . . . . . . . . . . . .  16
     9.2.  Informative References  . . . . . . . . . . . . . . . . .  20
   Appendix A.  Change Log . . . . . . . . . . . . . . . . . . . . .  20
   Appendix B.  Open Issues  . . . . . . . . . . . . . . . . . . . .  20
   Appendix C.  YANG Tree  . . . . . . . . . . . . . . . . . . . . .  20
   Appendix D.  Applicability examples . . . . . . . . . . . . . . .  23
   Authors' Addresses  . . . . . . . . . . . . . . . . . . . . . . .  24

Galimberti, et al.        Expires 21 June 2025                  [Page 2]

Internet-Draft   draft-ietf-ccamp-dwdm-if-param-yang-12    December 2024

1.  Introduction

   This memo defines a Yang model for managing single channel optical

>    interface parameters of coherent interfaces supporting DWDM

Reviewer: Is this model only applicable to coherent interfaces?

   applications, using the approach specified in G.698.2.  This model
   supports parameters to characterize coherent transceivers found in
   current implementations to specify the mode of operation.  As
   application identifiers like those specified in [ITU-T_G.874] and
   [ITU-T_G.698.2] may not always be available, mode templates are used.
   A mode template describes transceiver characteristics in detail and
   can be identified by a mode-id.

>    This draft refers and supports the [RFC7698] and is aligned to the

Author: Accepted. Resolved by changing the text based on this command proposal

>    definition of [RFC9093] and its extension RFC9093-bis.  Finally, the

Reviewer: RFC9093-bis should be replace with a normative reference to the I-D

Author: Deferring to next revision, Tracked with issue #11

   models described in this draft are compliant with the models
   described in draft-ietf-ccamp-optical-impairment-topology-yang and

>    draft-ietf-ccamp-wdm-tunnel-yang.

Reviewer: draft-ietf-ccamp-optical-impairment-topology-yang and draft-ietf-ccamp-wdm-tunnel-yang should be replaced with informative references to the corresponding I-Ds

Author: Deferring to next revision, Tracked with issue #11

   The key concept introduced by this YANG model in accordance with

>    RFC9093-bis is the notion of a mode.  A mode is a combination of

Author: Text updated as per comment suggestion

   parameters and parameter ranges that is supported by a transceiver.
   As an example, operating a device in QPSK modulation may use a
   different FEC and requires less OSNR than the same transceiver
   operating in QAM16 modulation.  Given the number of parameters and
   their possible combinations it is important for vendors to be able to
   qualify a set of combinations which is the basis to define a mode.
   As described in draft-ccamp-optical-impairment-topology-yang and
   modeled in RFC9093-bis, the list of transceiver modes (properly
   classified in standard, organizational and explicit modes) provide
   information about the transmission capabilities of an optical DWDM
   interface.  The same approach is used in this RFC to allow the device
   netconf agent to export such capabilities to the client controllers.

   To advertise the capability supported by an interface, a list of
   transceiver modes is provided by the device for each dwdm coherent
   module (supported-modes).

   The YANG model provides a configuration attribute named "current-
   mode" as a mean to provision the working mode of the interface from

>    that transceiver modes listed as capabilities.  Once provisioned,

Author: Text updated as per comment suggestion

   wdm-if-parameters provide the means to configure specific parameters
   at run time and retrieve configuration, performance and operational
   state information from the module.  For example, the frequency is a
   parameter that can be set within min/max boundaries defined in the
   current mode.  Laser Temperature, RX/TX optical power, RX OSNR, etc.
   however are read-only parameters available at run-time that can be
   checked against the mode boundaries and may trigger events.

Galimberti, et al.        Expires 21 June 2025                  [Page 3]

Internet-Draft   draft-ietf-ccamp-dwdm-if-param-yang-12    December 2024

2.  Conventions

   The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
   "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this
   document are to be interpreted as described in BCP 14 [RFC2119]
   [RFC8174] when, and only when, they appear in all capitals, as shown
   here.

> 3.  Narrative Section
> 
>    [Editor Note: Changed the section title according to draft-ietf-
>    netmod-rfc8407bis]
> 
> 3.1.  Overview

Reviewer: draft-ietf-netmod-rfc8407bis requires to have one or more narrative sections not to have a section with the tile "Narrative Section"

```suggestion
3.  Model Overview
```

Author: I would prefer to keep the Overview and other part of the textual description as sub-section of a common NArrative title. Agree on changing the title "Narrative Section" in something different, I propose "Module Description"

   Curr-q-factor:
      "Q" factor estimated at Rx Transceiver port.

3.4.  Use Cases

   The use cases are described in draft-ietf-ccamp-optical-impairment-
   topology-yang, draft-ietf-teas-actn-poi-applicability and draft-
   poidt-ccamp-actn-poi-pluggable-usecases-gaps.

> 4.  Definitions Section
> 
>    [Editor Note: Changed the section title according to draft-ietf-
>    netmod-rfc8407bis]
> 

Reviewer: draft-ietf-netmod-rfc8407bis requires to have one or more definition sections not to have a section with the tile "Definition Section"

```suggestion
```

> 4.1.  Optical Interface for external transponder in a WDM network
> 
>    [Editor Note: we may need to find a better title for this section and

Reviewer: ```suggestion
4.  WDM Interface YANG Module

   [Editor Note: we may need to find a better title for this section and
```

Author: Agree and updated as per comment

   The ietf-wdm-interface is an augment to the ietf-interface.  It
   allows the user to set the operating mode of transceivers as well as
   other operational parameters.  The module provides also threshold
   settings and notifications to supervise measured parameters and
   notify the client.

> 

Author: Better to defer this to the next revision after a discussion. opened issue #12 to track this comment

                 Italy
                 Sergio.belotti@nokia.com

               Italo Busi
                 Huawei Technologies

>                  Email: Italo.Busi@huawei.com

Author: Ok thanks
